Chessington.GameEngine/Pieces/Bishop.cs

```diff
@@ -10,7 +10,28 @@ public Bishop(Player player)
 
         public override IEnumerable<Square> GetAvailableMoves(Board board)
         {
-            return Enumerable.Empty<Square>();
+            List<Square> availableMoves = new List<Square>();
+            Square here = board.FindPiece(this);
+            foreach (var index in Enumerable.Range(1,7))
```

Comment:
You can also use 4 While loops, I think that will be neater for this whole loops